ui/app/src/components/ConfirmDialog/ConfirmDialog.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DispatchWithoutAction, ReactNode } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button } from '@mui/material';
+
+export interface ConfirmDialogProps {
+  title: ReactNode;
+  children?: ReactNode;
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: DispatchWithoutAction;
+}
+
+/**
+ * This is a simple and generic Confirm Dialog asking a question and answering by Yes or No.
+ * TODO: Move it to perses components or core components if it is designed to be used as a generic confirmation dialog
+ * TODO: This dialog is quite generic but maybe does not wrap enough some business logic that could help building
+ *  confirmation. Indeed all deletion are done in the same way in the API (restful) and it would be interesting to have
+ *  a component embedding this logic.
```

Comment:
I think these both will benefit from some deeper discussion before happening, especially if they start baking in any backend querying logic.

Reply:
This component might be too generic and does not offer enough flexibility for it to be a perses/core component. For example we might want more descriptive button labels. However i do think it might be good to start thinking about a reusable `Dialog` component to ensure a consistent dialog design

Reply:
Thanks for your feedback! yes I can probably think of creating such a component, but what would it contain? I'm not sure I have enough experience in the project to adjust the cursor properly between flexibility and consistency. 

My own experience in another UI we maintain is that we have a confirmation dialog which takes a list of action with that structure:
```ts
export class Action {
  kind: KindEnum;
  name: string;
  type: ActionTypeEnum;
  action: Observable<ActionResult>;
}
```
and then all the sentences in the dialog are created from that input, and action is performed if user click on submit.
This is very opinionated approach and only you can tell me if it would apply in Perses.

Even if we don't go that far, a text utilities could be useful at a start to make sure that if used, the same wording is displayed.

Reply:
Let me propose you something. Now that I renamed to the button from yes/no to delete/cancel, I think I understand better what you mean. 

**I will for now remove the TODO and rename it to DeleteProjectDialog.**

Then I will be free to check all current dialogs and propose something with a discussion thread + eventually a PR.

Reply:
Here is the discussion I created. Let's continue this thread here https://github.com/perses/perses/discussions/935